### PR TITLE
[multiple-choice] Simplify and use tokenizer.encode_plus

### PR DESCRIPTION
**Pull request by julien-c**
Our base tokenizer `PreTrainedTokenizer` now has the ability to encode a sentence pair up to a `max_length`, adding special tokens for each model and returning a mask of `token_type_ids`.

In this PR we upgrade `run_multiple_choice` by adopting this factorized tokenizer API.

To ensure the results are strictly the same as before, we implement a new `TruncatingStrategy` (ideally this could be an enum).

@erenup as you spent a lot of time on this script, would you be able to review this PR?

Result of eval with parameters from [examples/readme](https://github.com/huggingface/transformers/blob/julien_multiple-choice/examples/README.md#multiple-choice):
```
eval_acc = 0.8352494251724483
eval_loss = 0.42866929549320487
```

### CODE REVIEW

**Comment by LysandreJik** [comment]
Great addition. I feel like using enums would be especially helpful for the truncating strategy, indeed.

**Review by LysandreJik — APPROVED** [review]

File: examples/run_multiple_choice.py [hunk 1]
```diff
@@ -362,7 +362,7 @@ def main():
                         help="Whether to run eval on the dev set.")
     parser.add_argument("--do_test", action='store_true', help='Whether to run test on the test set')
     parser.add_argument("--evaluate_during_training", action='store_true',
-                        help="Rul evaluation during training at each logging step.")
+                        help="Run evaluation during training at each logging step.")
```
Comment (erenup) [inline examples/run_multiple_choice.py]: Could you please help to fix a `--overwrite_cache ` problem at line 296 of this file in this PR.
I think `line 296` should be changed to 
```
if os.path.exists(cached_features_file) and not args.overwrite_cache:
```
to make `--overwrite_cache` valid.
Since I cannot add a comment on line 296, I put it in the last change of this file. Thank you.

File: transformers/tokenization_utils.py [hunk 2]
```diff
@@ -825,21 +829,31 @@ def prepare_for_model(self, ids, pair_ids=None, max_length=None, add_special_tok
...
-                    elif not truncate_first_sequence and pair:
-                        encoded_inputs["overflowing_tokens"] = pair_ids[max_length - len_ids - n_added_tokens - stride:]
-                        pair_ids = pair_ids[:max_length - len_ids - n_added_tokens]
-                    else:
-                        logger.warning(
-                            "Cannot truncate second sequence as it is not provided. No truncation.")
+
+            if n_added_tokens + len_ids + len_pair_ids > max_length:
+                if truncate_both_sequences:
+                    tokens_a, tokens_b = self._truncate_seq_pair(
```
Comment (erenup) [inline transformers/tokenization_utils.py]: I'm wondering if we need to return `tokens_a`, `tokens_b` in `self._truncate_seq_pair`, though this line is able to work now.

**Comment by erenup** [comment]
I have reviewed this PR and It looks good to me. Thank you! @julien-c . I added two lines of comments above. Hope they are useful. Thank you.

File: examples/run_glue.py [hunk 3]
```diff
@@ -270,7 +270,7 @@ def load_and_cache_examples(args, task, tokenizer, evaluate=False):
         list(filter(None, args.model_name_or_path.split('/'))).pop(),
         str(args.max_seq_length),
         str(task)))
-    if os.path.exists(cached_features_file):
+    if os.path.exists(cached_features_file) and not args.overwrite_cache:
```
Comment (julien-c) [inline examples/run_glue.py]: Also added it here for glue @LysandreJik

**Comment by thomwolf** [comment]
Merged in, and superseded by, #1384
